ember-toucan-core/src/components/form/field.ts

```diff
@@ -0,0 +1,20 @@
+import Component from '@glimmer/component';
+
+import Wrapper from './wrapper';
+
+interface ToucanFormFieldComponentSignature {
+  Element: HTMLDivElement;
+  Args: {};
+  Blocks: {
+    // TODO - type these properly!
```

Comment:
We tried this:
```
interface ToucanFormWrapperComponentSignature {
  Element: HTMLDivElement;
  Args: {};
  Blocks: {
    default?: [
      {
        // (property) Label: typeof Label
        // Type 'abstract new () => PartiallyAppliedComponent<EmptyObject, {}, AcceptsBlocks<FlattenBlockParams<{ default: { Params: { Positional: []; }; }; }>, HTMLLabelElement>>' is not assignable to type 'typeof ToucanFormLabel'.
        // Cannot assign an abstract constructor type to a non-abstract constructor type.glint:ts(2322)
        Label: typeof Label;
        Hint: typeof Hint;
        Control: typeof Control;
        Error: typeof Error;
      }
    ];
// Error: Property '[Data]' is missing in type 'abstract new () => PartiallyAppliedComponent<EmptyObject, {}, AcceptsBlocks<FlattenBlockParams<{ default: { Params: { Positional: []; }; }; }>, HTMLDivElement>>' but required in type 'TemplateOnlyComponent<ToucanFormHintComponentSignature>'
    hint?: [component: typeof Hint];
    control?: [component: typeof Control];
    label?: [component: typeof Label];
    error?: [component: typeof Error];
  };
}
```

We were also getting an error

Reply:
If anyone knows how to fix those errors (and also: what's wrong), we're looking for advice!

Reply:
I think headless-form has examples of this: https://github.com/CrowdStrike/ember-headless-form/blob/main/packages/ember-headless-form/src/components/headless-form.ts#L50

Reply:
So we went and tried that, also following this guide: https://typed-ember.gitbook.io/glint/using-glint/ember/contextual-components.

But currently stuff like `Label` and `Hint` do not have any `args` so the guide recommends the approach above (just using `typeof Label`. Is it possible to book some time next week to pair on this?

Reply:
yeah, feel free to throw something on my calendar 🎉

Reply:
@nicolechung I can loop you in tomorrow, but after some discussion we got this fixed.  We ended up talking a bit more and decided to allow for named-blocks from components like `Input`, but `Field` being contextual may be a bit better for cases like Checkbox (since it breaks out of our normal vertical-flow).